lib/elixir/lib/enum.ex

```diff
@@ -4436,7 +4436,10 @@ defmodule Enum do
           end
 
         entry, {start, amount, to_drop, list} ->
-          {:halt, {start, amount, to_drop, [entry | list]}}
+          case to_drop do
+            1 -> {:halt, {start, amount, to_drop, [entry | list]}}
+            _ -> {:halt, {start, amount, to_drop, list}}
+          end
```

Comment:
My 2c:

```suggestion
          if to_drop == 1 do
            {:halt, {start, amount, to_drop, [entry | list]}}
          else
            {:halt, {start, amount, to_drop, list}}
          end
```

Reply:
Thank you for the comment. The idea here was to keep consistency with the code just above:

```elixir
case to_drop do
   1 -> {:cont, {start, amount - 1, step, [entry | list]}}
   _ -> {:cont, {start, amount - 1, to_drop - 1, list}}     
end
```

Do you think both should be changed? Or is there a performance implication behind the original code?

Reply:
Performance is the same. Both snippets are fine, so let’s keep it as is. :)